types/alpinejs/index.d.ts

```diff
@@ -269,7 +269,8 @@ export interface Alpine {
      * @param name the id of the x-data context
      * @param callback the initializer of the x-data context
      */
-    data(name: string, callback: (...initialStateArgs: unknown[]) => AlpineComponent): void;
+    // eslint-disable-next-line no-unnecessary-generics
+    data<TInitialStateArgs extends unknown[]>(name: string, callback: (...initialStateArgs: TInitialStateArgs) => AlpineComponent): void;
```

Comment:
Is the goal of this PR to improve assignability, or is there some other advantage to using a type parameter?

If it's just assignability, since the type parameter is basically unused, can't you type this as
`data(name: string, callback: (...initialStateArgs: any[]) => AlpineComponent): void;`
or
`data(name: string, callback: (...initialStateArgs: never[]) => AlpineComponent): void;`
?